internal/pkg/flink/internal/store/store_test.go

```diff
@@ -799,3 +815,414 @@ func TestTimeout(t *testing.T) {
 		require.Equal(t, tc.expected, result, tc.name)
 	}
 }
+
+func (s *StoreTestSuite) TestProcessStatement() {
+	client := mock.NewMockGatewayClientInterface(gomock.NewController(s.T()))
+	appOptions := &types.ApplicationOptions{
+		OrgResourceId:  "orgId",
+		EnvironmentId:  "envId",
+		ComputePoolId:  "computePoolId",
+		IdentityPoolId: "identityPoolId",
+	}
+	store := Store{
+		Properties: map[string]string{
+			"TestProp": "TestVal",
+		},
+		client:     client,
+		appOptions: appOptions,
+	}
+
+	statusDetailMessage := "Test status detail message"
+	statementObj := flinkgatewayv1alpha1.SqlV1alpha1Statement{
+		Status: &flinkgatewayv1alpha1.SqlV1alpha1StatementStatus{
+			Phase:  "PENDING",
+			Detail: &statusDetailMessage,
+		},
+	}
+
+	statement := "SELECT * FROM table"
+	client.EXPECT().CreateStatement("orgId",
+		"envId",
+		"computePoolId",
+		"identityPoolId",
+		statement,
+		store.propsDefault(store.Properties)).
+		Return(statementObj, nil)
+
+	processedStatement, err := store.ProcessStatement(statement)
+	require.Nil(s.T(), err)
+	require.Equal(s.T(), types.NewProcessedStatement(statementObj), processedStatement)
+}
+
+func (s *StoreTestSuite) TestProcessStatementFailsOnError() {
+	client := mock.NewMockGatewayClientInterface(gomock.NewController(s.T()))
+	appOptions := &types.ApplicationOptions{
+		OrgResourceId:  "orgId",
+		EnvironmentId:  "envId",
+		ComputePoolId:  "computePoolId",
+		IdentityPoolId: "identityPoolId",
+	}
+	store := Store{
+		Properties: map[string]string{
+			"TestProp": "TestVal",
+		},
+		client:     client,
+		appOptions: appOptions,
+	}
+
+	statusDetailMessage := "Test status detail message"
+	statementObj := flinkgatewayv1alpha1.SqlV1alpha1Statement{
+		Status: &flinkgatewayv1alpha1.SqlV1alpha1StatementStatus{
+			Detail: &statusDetailMessage,
+		},
+	}
+
+	statement := "SELECT * FROM table"
+	client.EXPECT().CreateStatement("orgId",
+		"envId",
+		"computePoolId",
+		"identityPoolId",
+		statement,
+		store.propsDefault(store.Properties)).
+		Return(statementObj, errors.New("Test error"))
+	expectedError := &types.StatementError{
+		Msg:            "Test error",
+		FailureMessage: "Test status detail message",
+	}
+
+	processedStatement, err := store.ProcessStatement(statement)
+	require.Nil(s.T(), processedStatement)
+	require.Equal(s.T(), expectedError, err)
+}
+
+func (s *StoreTestSuite) TestWaitPendingStatement() {
+	client := mock.NewMockGatewayClientInterface(gomock.NewController(s.T()))
+	appOptions := &types.ApplicationOptions{
+		OrgResourceId: "orgId",
+		EnvironmentId: "envId",
+	}
+	store := Store{
+		Properties: map[string]string{
+			"TestProp": "TestVal",
+		},
+		client:     client,
+		appOptions: appOptions,
+	}
+
+	statementName := "Test Statement"
+	statusDetailMessage := "Test status detail message"
+	statementObj := flinkgatewayv1alpha1.SqlV1alpha1Statement{
+		Spec: &flinkgatewayv1alpha1.SqlV1alpha1StatementSpec{
+			StatementName: &statementName,
```

Comment:
```suggestion
			StatementName: flinkgatewayv1alpha1.PtrString("Test Statement"),
```

Reply:
In this case I need statementName again in line 923, but I'm using `PtrString()` now wherever I don't need the variable more than once!